src/app/multi-upload/multi-upload.component.ts

```diff
@@ -21,35 +27,48 @@ export class MultiUploadComponent {
     this.addFileFromDragEvent(<DragEvent> event);
   }
 
+  removeFileBy(id: string, event: Event) {
+    event.preventDefault();
+    this.resultFiles = this.resultFiles.filter(file => file.id !== id);
+    this.ref.detectChanges();
+  }
+
   private addFileFromInputElement(uploadInput: HTMLInputElement) {
     const files = uploadInput.files;
     if (!files || files.length === 0) { return; }
-    this.addFiles(this.convertToArray(files));
+    this.addFiles(files);
   }
 
   private addFileFromDragEvent(dragEvent: DragEvent) {
-    this.addFiles(this.convertToArray(dragEvent.dataTransfer.files));
+    this.addFiles(dragEvent.dataTransfer.files);
   }
 
-  private addFiles(files: File[]) {
-    if (this.files) {
-      this.files = this.files.concat(files);
-    } else {
-      this.files = files;
+  private addFiles(files: FileList) {
+
+    for (const file of Array.from(files)) {
+      const resultFile = this.fileBuilderService.buildResult(file);
```

Comment:
I build the file here, so we can work with the result and don't need to worry about the files anymore